### PR TITLE
shop page with filter product by categories and price

### PR DESCRIPTION
**Pull request by songtv-0006**
Create shop page and filter product by categories, price low to high and high to low

### CODE REVIEW

File: resources/views/shop/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+						<div><a href="#" class="sorting mr-auto genric-btn default">@lang('High to Low') <i class="fas fa-arrow-circle-down"></i></a></div>
+						<div class="sorting mr-auto"></div>
+						{{ $products->appends(request()->input())->links() }}
+					</div>
+					<!-- End Filter Bar -->
+				</div>
+			</div>
+		</div>
+	</section>
+@endsection
```
Comment (duccs-0007) [inline resources/views/shop/index.blade.php:118]: endline

File: public/css/custom/custom.css [hunk 2]
```diff
@@ -27,3 +27,22 @@
...
+}
+
+.no_product {
+    padding-top: 20px;
+}
+
+.resize_product_item {
+    width: 255px !important;
+    height: 255px !important;
+}
```
Comment (duccs-0007) [inline public/css/custom/custom.css]: endline

File: app/Http/Controllers/ShopController.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+use Illuminate\Http\Request;
+use App\Models\Product;
+use App\Models\Category;
+
+class ShopController extends Controller
+{
+    public function index()
+    {
+        $count_product = Product::count();
+        $category_title = request()->category;
```
Comment (duccs-0007) [inline app/Http/Controllers/ShopController.php:14]: MassAssign ?

**Comment by songtv-0006** [comment]
ready

File: app/Http/Controllers/ShopController.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        $categories = Category::withCount('products')->get();
+        
+        if (request()->category)
+        {
+            if($category = Category::where('name', request()->category)->first())
+            {
+                $products = Product::where('category_id', $category->id);
```
Comment (tranducquoc) [inline app/Http/Controllers/ShopController.php]: dùng Eager-loading

**Comment by songtv-0006** [comment]
ready

**Pull request merged**